tests/api.cpp

```diff
@@ -59,133 +70,69 @@ int main( int argc, char** argv )
    try {
       fc::api<calculator> calc_api( std::make_shared<some_calculator>() );
 
-      fc::http::websocket_server server;
-      server.on_connection([&]( const websocket_connection_ptr& c ){
+      auto server = std::make_shared<fc::http::websocket_server>();
+      server->on_connection([&]( const websocket_connection_ptr& c ){
                auto wsc = std::make_shared<websocket_api_connection>(*c, MAX_DEPTH);
                auto login = std::make_shared<login_api>();
                login->calc = calc_api;
                wsc->register_api(fc::api<login_api>(login));
                c->set_session_data( wsc );
           });
 
-      server.listen( 8090 );
-      server.start_accept();
-
-      for( uint32_t i = 0; i < 5000; ++i )
-      {
-         try { 
-            fc::http::websocket_client client;
-            auto con  = client.connect( "ws://localhost:8090" );
-            auto apic = std::make_shared<websocket_api_connection>(*con, MAX_DEPTH);
-            auto remote_login_api = apic->get_remote_api<login_api>();
-            auto remote_calc = remote_login_api->get_calc();
-            remote_calc->on_result( []( uint32_t r ) { elog( "callback result ${r}", ("r",r) ); } );
-            wdump((remote_calc->add( 4, 5 )));
-         } catch ( const fc::exception& e )
-         {
-            edump((e.to_detail_string()));
-         }
-      }
-      wlog( "exit scope" );
-   } 
-   catch( const fc::exception& e )
-   {
-      edump((e.to_detail_string()));
-   }
-   wlog( "returning now..." );
-   
-   return 0;
+      server->listen( 8090 );
```

Comment:
Please find a random free port and use that. I believe there's code in `cli_test` that does it too.